internal/exec/engine.go

```diff
@@ -76,6 +76,14 @@ func (e Engine) Run(stageName string) bool {
 
 	e.Logger.PushPrefix(stageName)
 	defer e.Logger.PopPrefix()
+
+	if e.Root != "/" {
+		baseConfig.Storage.Filesystems = []types.Filesystem{{
+			Name: "root",
+			Path: func(p string) *string { return &p }(e.Root),
```

Comment:
I recently made an `IntToPtr` in `internal/util/util.go`, could you make a `StringToPtr` there and use it here?